src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -1,26 +1,95 @@
 #include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
 
+#ifdef MIXXX_USE_QML
+#include <QDirIterator>
+#include <QQmlEngine>
+#include <QQuickItem>
+#include <QQuickWindow>
+#include <QtEndian>
+#include <algorithm>
+
+// Prevent conflict with methods called 'emit' in <execution> source
+#pragma push_macro("emit")
+#undef emit
+#include <execution>
+#pragma pop_macro("emit")
+#endif
+
 #include "control/controlobject.h"
 #include "controllers/controller.h"
+#ifdef MIXXX_USE_QML
+#include "controllers/rendering/controllerrenderingengine.h"
+#endif
 #include "controllers/scripting/colormapperjsproxy.h"
 #include "controllers/scripting/legacy/controllerscriptinterfacelegacy.h"
 #include "errordialoghandler.h"
 #include "mixer/playermanager.h"
 #include "moc_controllerscriptenginelegacy.cpp"
+#ifdef MIXXX_USE_QML
+#include "util/assert.h"
+#include "util/cmdlineargs.h"
+
+using Clock = std::chrono::steady_clock;
+#endif
+
+#ifdef MIXXX_USE_QML
+namespace {
+const QByteArray kScreenTransformFunctionUntypedSignature =
+        QMetaObject::normalizedSignature(
+                "transformFrame(QVariant,QVariant)");
+const QByteArray kScreenTransformFunctionTypedSignature =
+        QMetaObject::normalizedSignature("transformFrame(QVariant,QDateTime)");
+const QByteArray kScreenInitFunctionUntypedSignature =
+        QMetaObject::normalizedSignature(
+                "init(QVariant,QVariant)");
+const QByteArray kScreenInitFunctionTypedSignature =
+        QMetaObject::normalizedSignature("init(QString,bool)");
+const QByteArray kScreenShutdownFunctionSignature =
+        QMetaObject::normalizedSignature("shutdown()");
```

Comment:
QML support typing annotations. If the function contains signature annotations, the Qt type inferred from it will be used instead of `QVariant`. Having both typed and untyped allows support for both.

Reply:
I see, thats new. I didn't know that was a feature.